server/master/src/main/java/org/apache/accumulo/master/upgrade/UpgradeCoordinator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.master.upgrade;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.ServerUtil;
+
+public interface UpgradeCoordinator {
```

Comment:
It's not clear that we need this interface. I like the returning of the NOOP coordinator... but I think the same could be achieved inside the single UpgradeCoordinatorImpl. The "needsupgrade" check with the ServerUtil could occur inside the constructor instead of outside in the factory "create" method.

Reply:
I collapsed it in 46e91f5417e33a5fa9f64345bf8497cc3205ce95